src/component/ViewInvoice/index.jsx

```diff
@@ -1,6 +1,104 @@
-function ViewInvoice() {
+import { useParams } from "react-router-dom"
+import ViewHeader from "../ViewHeader"
+import { useEffect, useState } from "react"
+import moment from "moment"
+import Bill from "../Bill"
+import StatusIcon from "../StatusIcon"
+
+function ViewInvoice({ }) {
+    const { idName, id } = useParams()
+    const [clientInvoice, setClientInvoice] = useState({
+        name: "",
+        streetAddress: "",
+        city: "",
+        created: "",
+        due: "",
+        details: [],
+        paidToDate: "",
+        invoiceTotal: "",
+        statusName: ""
+    })
+    const { name, streetAddress, city, created, due, details, paidToDate, invoiceTotal, statusName } = clientInvoice
+    const date = moment(due)
+    useEffect(() => {
+        fetch(`https://invoicing-api.dev.io-academy.uk/invoices/${id}`)
+            .then(res => res.json())
+            .then((invoiceData) => {
+                let data = invoiceData.data
+                setClientInvoice({
+                    name: data.name
+                    , streetAddress: data.street_address
+                    , city: data.city
+                    , created: data.created
+                    , due: data.due
+                    , details: data.details
+                    , paidToDate: data.paid_to_date
+                    , invoiceTotal: data.invoice_total
+                    , status: data.status
+                    , statusName: data.status_name
+                })
+            })
+    }, [])
+
+    const dateDue = moment(due);
+    const formattedDateDue = dateDue.format("DD MMMM YYYY");
+    const dateCreated = moment(created);
+    const formattedDateCreated = dateCreated.format("DD MMMM YYYY");
```

Comment:
You should do this in a useEffect and make state for the data you're using in your JSX. This code will run way to often outside useEffect

Reply:
BTW, you should put this in it's own useEffect rather than adding it into the existing one